[yingx9] iP

yingx9 opened the pull request.

yingx9 committed:
Add Level-0: Rename, Greet, and Exit

yingx9 committed:
Add Level-1: Echo

yingx9 committed:
Add Level-1: Echo

yingx9 committed:
Add Level-2: Add, List

yingx9 committed:
Add Level-3: Mark as Done

yingx9 committed:
Add A-CodingStandard: Follow the Coding Standard

yingx9 committed:
Add Level-4: ToDo, Event, Deadline and A-Inheritance

yingx9 committed:
Rename Duke.java and edit for A-CodeQuality: Improve Code Quality

Magpie.java: extract a method markTask

Marking and unmarking share some common code behaviors

Hence, new method "markTask" was created to improve code quality

runtest.bat,EXPECTED.TXT, input.txt were edited to implement
Automated Text UI testing

DavinciDelta commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,40 @@
+        return description;
+    }
+
+    public void setDescription(String description) {
```
I like the use of camelCasing and the boolean variables have great names! but the setDescription seems to be redundant as of right now, maybe use it for the constructor?

DavinciDelta commented on src/main/java/Magpie.java:
```diff
@@ -0,0 +1,131 @@
+            String arguments = input.substring(command.length());
+
+
+            switch(command){
```
I like that the switch statements have the correct formatting and indentation! but switch case in a while looks very confusing, maybe put the switch in another java class?

DavinciDelta commented on src/main/java/Magpie.java:
```diff
@@ -0,0 +1,131 @@
+
+    }
+
+}
```
I like the indentations!

DavinciDelta commented:
very good code!

yingx9 committed:
Add classes for input and task handling and edit for error handling

Created classes: inputHandler, inputValidator, and taskHandler

The switch case in Magpie.java was becoming long and inefficient

Extracting it into classes, as well as breaking it up into methods
improves readability.

Error handling was implemented in inputValidator and taskHandler

yingx9 committed:
Merge branch 'branch-Level-5'

yingx9 committed:
Add MagpieException for A-Exceptions and edit inputValidator

Validating methods in inputValidator now throws MagpieException,
which will be caught and handled to print specific error messages

yingx9 committed:
Sorted classes into different packages for A-Packages

yingx9 committed:
Merge branch 'branch-A-Packages'

yingx9 committed:
Add Level-6: Delete and edit array to use ArrayList Collections

Primitive data structures like a normal array are not as flexible
as java Collection classes

Hence Task Array was modified to a ArrayList data structure

deleteTask method was also added to implement Level-6

Musfirahe0556596 commented on src/main/java/Magpie.java:
```diff
@@ -0,0 +1,48 @@
+    private static final String LINE = "\n____________________________________________________________";
+
+
+    public static void main(String[] args) {
+
+
+        System.out.println(LINE);
+        System.out.println(" __  __          _____ _____ _____ ______ ");
+        System.out.println("|  \\/  |   /\\   / ____|  __ \\_   _|  ____|");
+        System.out.println("| \\  / |  /  \\ | |  __| |__) || | | |__   ");
+        System.out.println("| |\\/| | / /\\ \\| | |_ |  ___/ | | |  __|  ");
+        System.out.println("| |  | |/ ____ \\ |__| | |    _| |_| |____ ");
+        System.out.println("|_|  |_/_/    \\_\\_____|_|   |_____|______|");
+        System.out.println("\nHello! I'm Magpie :)");
+        System.out.println("What can I do for you?\n");
+        System.out.println(LINE);
+
+
+        Scanner in = new Scanner(System.in);
+        String input = "";
+        input = in.nextLine();
+
+        taskHandler taskManager = new taskHandler();
+
+        while (!input.equalsIgnoreCase("bye")) {
+
+            inputHandler userInput = new inputHandler(input);
+            try{
+                userInput.processCommand(taskManager);
+            }
+            catch (MagpieException e){
+                System.out.println(e.getErrorMessage());
+            }
+            input = in.nextLine();
+
+
+        }
+        System.out.println(LINE);
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println(LINE);
```
Avoid long methods (goes beyond 30 LoC)

Musfirahe0556596 commented on src/main/java/taskHandler.java:
```diff
@@ -0,0 +1,73 @@
+        taskCount = 0;
+    }
+
+    public static void print() {
```
Naming of methods could be improved to describe what exactly this method prints

Musfirahe0556596 commented on src/main/java/taskHandler.java:
```diff
@@ -0,0 +1,73 @@
+        }
+    }
+
+    public static void add(Task t) {
```
Naming of methods could be improved to describe what is added

Musfirahe0556596 commented on src/main/java/inputValidator.java:
```diff
@@ -0,0 +1,55 @@
+public class inputValidator {
+
+    public static void ValidateTodo() throws MagpieException {
+
+        if (inputHandler.splitInput.length <= 1) {
+            throw new MagpieException("Please provide a description for your task!");
+        }
+    }
+
+    public static void ValidateDeadline() throws MagpieException {
+        boolean hasBy = inputHandler.arguments.contains("/by");
+
+        if (!hasBy) {
+            throw new MagpieException("Please specify a deadline using /by!");
+        }
+    }
+
+    public static void ValidateEvent() throws MagpieException {
+        boolean hasFrom = inputHandler.arguments.contains("/from");
+        boolean hasTo = inputHandler.arguments.contains("/to");
+
+        if(!hasFrom && !hasTo){
+            throw new MagpieException("Please specify starting and ending times using /from and /to!");
+        }
+        else if (!hasFrom) {
+            throw new MagpieException("Please specify a starting time using /from!");
+        }
+        else if (!hasTo) {
+            throw new MagpieException("Please specify a ending time using /to!");
+
+        }
+
+
+
+    }
+
+    public static void ValidateMarking() throws MagpieException {
```
Method name should be in camelCase

Musfirahe0556596 commented on src/main/java/taskHandler.java:
```diff
@@ -0,0 +1,73 @@
+
+    }
+
+    public static void mark(int index, boolean isDone) {
```
Naming of methods could be improved to describe what is marked

Musfirahe0556596 commented on src/main/java/inputHandler.java:
```diff
@@ -0,0 +1,100 @@
+
+    protected static String command;
+    protected static String arguments;
+    protected static String[] splitInput;
```
Plural form should be used on names representing a collection of objects.

Musfirahe0556596 reviewed (commented):
Overall looks good but there are some coding standards and quality to be followed!

yingx9 committed:
Add fileHandler for Level-7: Save and edit for code quality

Tasks should be saved to the hard disk when added, deleted or
changed

Hence, lets:
-Create a fileHandler class to handle file-related methods
-Create a data folder for data.txt
-Check if data.txt exists and create one if it doesn't exist
-Update file whenever task is modified by calling fileHandler
methods in taskHandler

Feedback from peers was taken into account and various method names
were edited for code quality

yingx9 committed:
Merge branch 'branch-Level-6'

yingx9 committed:
Merge branch 'branch-Level-7'

yingx9 committed:
Update fileHandler to create directory path if it does not exist

Previously the path was absolute and would lead to errors

Hence fileHandler was edited to prevent errors were the path does
not exist

Minor updates were made to method names to adhere to coding
standards

yingx9 committed:
Refactor code to extract classes for A-MoreOOP

Rename inputHandler, fileHandler, and taskHandler to Parser,
Storage, and TaskList respectively to reflect naming given in
A-MoreOOP

Extracted user interaction code into new Ui class

yingx9 committed:
Add findTask method for Level-9: Find

yingx9 committed:
Add JavaDoc comments for classes and methods for A-JavaDoc

yingx9 committed:
Merge pull request #1 from yingx9/branch-Level-9

Add findTask method for Level-9: Find

yingx9 committed:
Merge pull request #2 from yingx9/branch-A-JavaDoc

Add Java Doc comments

yingx9 committed:
Tidy up code to follow coding standard

yingx9 committed:
Update file method to be more cross-platform compatible

Issue with updating data.txt found when testing on Linux

Hence, update taskFromFile to use System.lineSeparator() to
be more cross-platform compatible

yingx9 committed:
Update README.md

yingx9 committed:
Update README.md

yingx9 committed:
Update README.md

yingx9 committed:
Update README.md for A-UserGuide

yingx9 committed:
Update README.md to fix reference links

yingx9 committed:
Rename inputValidator.java to InputValidator.java

yingx9 committed:
Update Storage.java to follow coding standard
